cmd/waku/server/rest/store.go

```diff
@@ -198,13 +199,9 @@ func (d *StoreService) getV1Messages(w http.ResponseWriter, r *http.Request) {
 
 	ctx, cancel := context.WithTimeout(r.Context(), 5*time.Second)
 	defer cancel()
-
-	_, err = d.node.AddPeer(peerAddr, peerstore.Static, d.node.Relay().Topics())
-	if err != nil {
-		writeStoreError(w, http.StatusInternalServerError, err)
-		return
+	if peerAddr != nil {
+		options = append(options, store.WithPeerAddr(peerAddr))
```

Comment:
I just moved it down from REST API thinking this could be a useful addition for all API's.
Yeah, created an issue #894 for having this option for other protocols.